src/config.ts

```diff
@@ -748,6 +748,34 @@ export default {
   } as Partial<Record<string, string[]>>,
   /* tslint:enable no-object-literal-type-assertion */
 
+  /**
+   * The Manifest parsing logic has a notion of "unsafeMode" which allows to
+   * speed-up this process a lot with a small risk of de-synchronization with
+   * what actually is on the server.
+   * Because using that mode is risky, and can lead to all sort of problems, we
+   * regularly should fall back to a regular "safe" parsing every once in a
+   * while.
+   * This value defines how many consecutive time maximum the "unsafeMode"
+   * parsing can be done.
+   */
+  MAX_CONSECUTIVE_MANIFEST_PARSING_IN_UNSAFE_MODE: 10,
+
+  /**
+   * Minimum time spent parsing the Manifest before we can authorize parsing
+   * it in an "unsafeMode", to speed-up the process with a little risk.
+   * Please note that this parsing time also sometimes includes idle time such
+   * as when the parser is waiting for a request to finish.
+   */
+  MIN_MANIFEST_PARSING_TIME_TO_ENTER_UNSAFE_MODE: 200,
+
+  /**
+   * Minimum amount of <S> elements in a DASH MPD's <SegmentTimeline> element
+   * necessary to begin parsing the current SegmentTimeline element in an
+   * unsafe manner (meaning: with risks of de-synchronization).
+   * This is only done when the "unsafeMode" parsing mode is enabled.
+   */
+  MIN_DASH_S_ELEMENTS_TO_PARSE_UNSAFELY: 300,
```

Comment:
How can we be sure that there will be always so many S elements ?

Reply:
I considered that for less than 300 elements, the gain of "smart-skipping" elements was negligible here (though it was a little random, it does not correspond to a real-life tests).

I also chose to prefer disabling this logic when it might be needed than the opposite (false negatives over false positives).

--

In the realized tests we were having in about the tens of thousands of S elements, which seems far away from 300.